contracts/modules/common/LimitManager.sol

```diff
@@ -113,115 +81,62 @@ abstract contract LimitManager is BaseModule {
-                _unspent = limit - expense.alreadySpent;
-            }
+            lStorage.setDailySpent(_wallet, safe128(_amount), safe64(now + 24 hours));
+            return true;
+        } else if (periodEnd > now && alreadySpent.add(_amount) <= currentLimit) {
+            lStorage.setDailySpent(_wallet, safe128(alreadySpent.add(_amount)), safe64(periodEnd));
+            return true;
         }
+        return false;
     }
 
     /**
-    * @dev Helper method to check if a transfer is within the limit.
-    * If yes the daily unspent for the current period is updated.
+    * @dev Checks if a transfer is within the limit. If yes the daily spent is not updated.
     * @param _wallet The target wallet.
     * @param _amount The amount for the transfer
+    * @return true if the transfer is withing the daily limit.
     */
-    function checkAndUpdateDailySpent(address _wallet, uint _amount) internal returns (bool) {
-        if (_amount == 0)
-            return true;
-        Limit storage limit = limits[_wallet].limit;
-        uint128 current = currentLimit(limit.current, limit.pending, limit.changeAfter);
-        if (isWithinDailyLimit(_wallet, current, _amount)) {
-            updateDailySpent(_wallet, current, _amount);
+    function checkDailySpent(address _wallet, uint256 _amount) internal view returns (bool) {
+        (uint256 current, uint256 pending, uint256 changeAfter, uint256 alreadySpent, uint256 periodEnd) = getLimitAndDailySpent(_wallet);
+        uint256 currentLimit = currentLimit(current, pending, changeAfter);
+        if (currentLimit == LIMIT_DISABLED) {
             return true;
         }
-        return false;
+        // solium-disable-next-line security/no-block-members
+        if (periodEnd < now) {
+            return (_amount <= currentLimit);
+        } 
+        return (alreadySpent.add(_amount) <= currentLimit);
     }
 
     /**
-    * @dev Helper method to update the daily spent for the current period.
+    * @dev Gets from storage the limit object of a wellet.
     * @param _wallet The target wallet.
-    * @param _limit The current limit for the wallet.
-    * @param _amount The amount to add to the daily spent.
+    * @return the limit object.
     */
-    function updateDailySpent(address _wallet, uint128 _limit, uint _amount) internal {
-        if (_limit != LIMIT_DISABLED) {
-            DailySpent storage expense = limits[_wallet].dailySpent;
-            // solium-disable-next-line security/no-block-members
-            if (expense.periodEnd < now) {
-                // solium-disable-next-line security/no-block-members
-                expense.periodEnd = uint64(now + 24 hours);
-                expense.alreadySpent = uint128(_amount);
-            } else {
-                expense.alreadySpent += uint128(_amount);
-            }
-        }
+    function getLimit(address _wallet) internal view returns (uint256, uint256, uint256) {
+        return lStorage.getLimit(_wallet);
```

Comment:
I found the combination of this `getLimit` and `currentLimit` functions confusing. It appears that `getLimit` does not give me back my wallet's actual limit if the securityPeriod has passed but for that I need to call `currentLimit`. I suggest we only have one getter for limit which gives us the currently active wallet limit. If we made the `limits` mapping public  whoever is interested in the actual values can query that.

Reply:
I've removed the `getLimit` method from the `LimitUtils` library.